#1

Ideas that can be examined and tested statistically must have statistics support, ideas that cannot be examined and tested statistically are subjective opinions and unable to be measured to statistics to analyse.
The key "criteria" defining a good null hypothesis is that the hypothesis is not a subjective opinion and has enough statistics to analyse. 
The difference between a null hypothesis and an alternative hypothesis is that a null hypothesis assumes that nothing happens but the alternative hypothesis assumes the opposite which is something happens. 

#2

When we are doing hypothesis testing, we are basically doing an interpretation of the population. Specifically, the null and alternative hypothesis, which are statements that we are assuming and testing, are all about parameters of the population, not the sample statistic. As a result, the result of hypothesis test that we get is an interpretation of the population. For example, if we have a null hypothesis of 'μ=15', that means we assume that the population mean is 15, and the result of hypothesis test decides weather or not this assumption is correct. An example of an invalid parameter would be a null hypothesis of 'x bar=15', this is invalid because x bar is the sample mean which is a sample statistic. Xi, as data in the sample, would not be a valid parameter for hypothesis. μ0, as the expected population mean, is a popultion parameter. 

#3

We imagine a world where the null hypothesis is true because we want to test how much variation is in the sampling distribution. When we are calculating the p-value,we also assume the null hypothesis is true, because we are trying to find if we can prove that the null hypothesis is wrong when are testing reject ot not reject the null hypothesis. For example, in the video we assumed that the dog is innocent because we are trying to test how much likely that it is not innocent.

#4

A smaller p-value makes the null hypothesis look more ridiculous because the p-value is the probability of  a result as extreme as the one we found in our sample. For example, in the video we assumed that the dog is innocent, and the p-value is the probability that the dog is innocent, when the p-value is smaller, it is more unlikely that the dog is innocent, and makes us to reject the null hypothesis and to conclude the alternative hypothesis, which is that the dog is not innocent.

#5

H0: P=0.5 (null hypothesis: the probability of tilting the head to the right is 0.5);
Ha: P/=0.5 (alternative hypothesis: the probability of tilting the head to the right is not 0.5).

In [7]:
import numpy as np
n = 10000 
x = 124  
xi = 80  
simulation = np.random.binomial(n=x, p=0.5, size=n)
p_value = np.mean(simulation >= xi)
print("The P-value is:", p_value)

The P-value is: 0.0008


#5

Conclusion:
The P-value we get is 0.0008, which is smaller than 0.001, therefore, according to the strength of evidence table, we have very strong evidence against the null hypothesis, meaning that the probability of people tilting the head to the right when kissing is not 0.5.

#6

A smaller p-value cannot definitively prove that the null hypothesis is false. We don't directly judge whether the p-value is large or small, we usually compare it to a predetermined value to determine if the evidence is strong enough. For example, im the last question, we used the strength of evidence table, which compare the p-value to different range, to determine our conclusion.
The p-value cannot definitively prove innocence or guilt because there is always the possibility of making either a Type I error (when we reject the null hypothesis even though it is actually true) or Type II error (when we fail to reject the null hypothesis even though it is false). For example, even though the chance of a neighbor causing a mess is small, it doesn't mean it can't happen.

In [13]:
#7

import numpy as np
import pandas as pd
import plotly.figure_factory as ff
import plotly.graph_objects as go

patient_data = pd.DataFrame({
    "PatientID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Age": [45, 34, 29, 52, 37, 41, 33, 48, 26, 39],
    "Gender": ["M", "F", "M", "F", "M", "F", "M", "F", "M", "F"],
    "InitialHealthScore": [84, 78, 83, 81, 81, 80, 79, 85, 76, 83],
    "FinalHealthScore": [86, 86, 80, 86, 84, 86, 86, 82, 83, 84]
})

patient_data['HealthScoreChange'] = patient_data['FinalHealthScore'] - patient_data['InitialHealthScore']

np.random.seed(1)
number_of_simulations = 10000
IncreaseProportionSimulations_underH0random = np.zeros(number_of_simulations)

for i in range(number_of_simulations):
    random_improvement = np.random.choice([0, 1], size=len(patient_data), replace=True)
    IncreaseProportionSimulations_underH0random[i] = random_improvement.mean()

population_parameter_value_under_H0 = 0.5
observed_statistic = (patient_data.HealthScoreChange > 0).mean()

SimStats_as_or_more_extreme_than_ObsStat = IncreaseProportionSimulations_underH0random >= observed_statistic
p_value = SimStats_as_or_more_extreme_than_ObsStat.sum() / number_of_simulations

print("Number of Simulations: ", number_of_simulations)
print("Observed Statistic: ", observed_statistic)
print("p-value: ", p_value)

hist_data = [IncreaseProportionSimulations_underH0random + np.random.uniform(-0.05, 0.05, size=len(IncreaseProportionSimulations_underH0random))]
group_labels = ['Simulated Sampling Distribution']
fig = ff.create_distplot(hist_data, group_labels, curve_type='normal', show_hist=True, show_rug=False, bin_size=0.1)

fig.add_shape(type="line", x0=observed_statistic, y0=0, x1=observed_statistic, y1=2.5,
              line=dict(color="Green", width=4), name="Observed Statistic")
fig.add_trace(go.Scatter(x=[observed_statistic], y=[2.75], text=["Observed<br>Statistic"], mode="text", showlegend=False))

fig.update_layout(
    title="Simulated Sampling Distribution under H0",
    xaxis_title="Mean Health Score Change", yaxis_title="Density"
)
fig.show()

Number of Simulations:  10000
Observed Statistic:  0.8
p-value:  0.0565


#7

Switching from a two-tailed to a one-tailed test changes the interpretation to only looking for a significant difference in one direction instead of both directions. We should expect the p-value to be smaller in a one-tailed test because it only considers one extreme side, not both sides.

In our conversation, you shared a Python code that performs a two-tailed test, and you asked for guidance on how to modify it for a one-tailed test. I explained that for a **right-tailed test**, you need to check if the observed statistic is greater than or equal to the simulated values, and for a **left-tailed test**, you check if the observed statistic is less than or equal to the simulated values.

To make the code more flexible, I suggested using a `test_type` parameter that allows you to switch between **left-tailed**, **right-tailed**, and **two-tailed** tests, all within the same code structure. This way, you can choose the type of test dynamically, without needing to write separate conditions for each.
https://chatgpt.com/share/670f3672-c788-8001-8fff-281624709c35

#8

H0: P=0.5 (null hypothesis: students are guessing randomly with a 50% chance of correctly identifying the tea);
Ha: P>0.5 (alternative hypothesis: students are better than random guessing and can correctly identify which was poured first more than 50% of the time).

In [16]:
import numpy as np

n = 80
x = 49
observed_proportion = x / n

p0 = 0.5

np.random.seed(1)
n_bootstraps = 10000
bootstrap_proportions = np.zeros(n_bootstraps)

for i in range(n_bootstraps):
    bootstrap_sample = np.random.binomial(n, p0)
    bootstrap_proportions[i] = bootstrap_sample / n

p_value = np.mean(bootstrap_proportions >= observed_proportion)

print(f"Observed proportion: {observed_proportion}")
print(f"Bootstrap p-value (right-tailed test): {p_value}")

Observed proportion: 0.6125
Bootstrap p-value (right-tailed test): 0.0291


#8

Given that the bootstrap p-value is 0.0291, which is less than the significance level of 0.05, therefore, we reject the null hypothesis, and have enough evidence to conclude the alternative hypothesis. This means there is significant evidence to suggest that the proportion of students who can correctly identify whether the tea or milk was poured first is greater than random guessing (P > 0.5). In this context, it means that the STA130 students likely have the ability to distinguish the order of pouring tea and milk better than what would be expected by chance alone.

#9

Yes.